src/Control/Monad/Bayes/Population.hs

```diff
@@ -210,6 +212,31 @@ resampleMultinomial ::
   Population m a
 resampleMultinomial = resampleGeneric multinomial
 
+-- | Only use the given resampler when the effective sample size is below a certain threshold
+onlyBelowEffectiveSampleSize ::
+  MonadDistribution m =>
+  -- | The threshold under which the effective sample size must fall before the resampler is used.
+  --   For example, this may be half of the number of particles.
+  Double ->
+  -- | The resampler to user under the threshold
+  (MonadDistribution m => Population m a -> Population m a) ->
+  -- | The new resampler
+  (Population m a -> Population m a)
+onlyBelowEffectiveSampleSize threshold resampler pop = do
+  ess <- lift $ effectiveSampleSize pop
```

Comment:
I think this is wrong because it will execute the `m` effects of `pop` again.

Reply:
It might be better to have a function `withEffectiveSampleSize :: Functor m => Population m a -> m (a, Double)`